### PR TITLE
Update PostgreSQL driver to 42.2.23

### PR DESCRIPTION
**Pull request by electrum**
The PostgreSQL connector is not yet updated due to incompatibilities.

### CODE REVIEW

**Review by findepi — CHANGES_REQUESTED** [review]
So we're upgrading ... without upgrading?
Looks like the new version is not being picked outside of tests

**Comment by dain** [comment]
Related to https://github.com/trinodb/trino/issues/7823

**Comment by electrum** [comment]
Yes, this updates usages for tests, and other future usages such as DB-based resource groups (which has been started in #8530). CockroachDB (#8317) is another usage that requires a newer driver.

Additionally, this adds a comment to the PostgreSQL connector that explains why the driver cannot be updated, which was attempted in #7241 due to CVEs in the older driver version.

**Review by dain — APPROVED** [review]

File: plugin/trino-postgresql/pom.xml [hunk 1]
```diff
@@ -81,6 +81,12 @@
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
+            <!--
+            PostgreSQL connector does not work with newer drivers due to
+            fixes to handle of multidimensional arrays (>= 42.2.14) and
+            numeric types (>= 42.2.11) (Trino #7823)
```
Comment (hashhar) [inline plugin/trino-postgresql/pom.xml:87]: Use a link that we can click instead. And maybe make this a TODO.

```suggestion
            numeric types (>= 42.2.11) (https://github.com/trinodb/trino/issues/7823)
```

**Review by hashhar — COMMENTED** [review]
I'm not sure what this benefits since the only thing using the driver directly in tests is testcontainers and queries that are executed directly on PostgreSQL. Trino would still be using the older driver even within tests (or does it not?)

**Comment by findepi** [comment]
> Yes, this updates usages for tests, and other future usages such as DB-based resource groups (which has been started in #8530). CockroachDB (#8317) is another usage that requires a newer driver.

Looks like we're solving a problem that we do not have yet.

> Additionally, this adds a comment to the PostgreSQL connector that explains why the driver cannot be updated, which was attempted in #7241 due to CVEs in the older driver version.

I agree that it is a good idea to have a comment!

**Pull request merged**